clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -528,20 +531,39 @@ object GarbageCollector {
   ) = {
     val reportLogsDst = concatToGCLogsPrefix(storageNSForHadoopFS, "summary")
     val reportExpiredDst = concatToGCLogsPrefix(storageNSForHadoopFS, "expired_addresses")
+    val deletedObjectsDst = concatToGCLogsPrefix(storageNSForHadoopFS, "deleted_objects")
+
     val time = DateTimeFormatter.ISO_INSTANT.format(java.time.Clock.systemUTC.instant())
     writeParquetReport(commitsDF, reportLogsDst, time, "commits.parquet")
-    writeParquetReport(expiredAddresses, reportExpiredDst, time)
+    try {
+      writeParquetReport(expiredAddresses, reportExpiredDst, time)
+      val expiredDF = spark.read.parquet(f"${reportExpiredDst}/dt=${time}/")
+      println(f"Total expired addresses: ${expiredDF.count()}")
+    } catch {
+      // This exception is thrown when there are no expired addresses
+      case e: Throwable => {
+        println("Error when trying to get expired addresses count, moving on:")
+        e.printStackTrace()
+      }
+    }
     writeJsonSummary(configMapper, reportLogsDst, removed.count(), gcRules, time)
-
     removed
       .withColumn(MARK_ID_KEY, lit(markID))
       .withColumn(RUN_ID_KEY, lit(runID))
       .write
       .partitionBy(MARK_ID_KEY, RUN_ID_KEY)
       .mode(SaveMode.Overwrite)
-      .parquet(
-        concatToGCLogsPrefix(storageNSForHadoopFS, s"deleted_objects/$time/deleted.parquet")
-      )
+      .parquet(f"${deletedObjectsDst}/dt=$time")
+    try {
+      val deleteDF = spark.read.parquet(f"${deletedObjectsDst}/dt=$time")
+      println(f"Total objects deleted (or already had been deleted): ${deleteDF.count()}")
```

Comment:
line 549 counts removed objects, can we use it instead of reading deleted and counting again?

Reply:
Nice catch, will do